src/Connector.SqlServer/Connector/SqlServerConnector.cs

```diff
@@ -253,16 +243,54 @@ public async Task VerifyExistingContainer(ExecutionContext executionContext, Str
             }
 
             await Task.WhenAll(tasks);
+
+            async Task createTable(SanitizedSqlName tableName, IEnumerable<ConnectionDataType> columns, IEnumerable<string> keys,
+                string context)
+            {
+                try
+                {
+                    IEnumerable<SqlServerConnectorCommand> commands = _features
+                        .GetFeature<IBuildCreateContainerFeature>()
+                        .BuildCreateContainerSql(executionContext, providerDefinitionId, schema, tableName, columns, keys, _logger)
+                        .ToList();
+
+                    var indexCommands = _features.GetFeature<IBuildCreateIndexFeature>()
+                        .BuildCreateIndexSql(executionContext, providerDefinitionId, schema, tableName, keys, _logger);
+                    commands = commands.Union(indexCommands);
+
+                    foreach (var command in commands)
+                    {
+                        _logger.LogDebug("Sql Server Connector - Create Container[{Context}] - Generated query: {sql}",
+                            context, command.Text);
+
+                        await _client.ExecuteCommandAsync(config, command.Text, command.Parameters);
+                    }
+                }
+                catch (Exception e)
+                {
+                    var message = $"Could not create Container {tableName} for Connector {providerDefinitionId}";
+                    _logger.LogError(e, message);
+                    throw new CreateContainerException(message);
+                }
+            }
         }
 
         public override async Task EmptyContainer(ExecutionContext executionContext, Guid providerDefinitionId,
             string id)
         {
             var config = await base.GetAuthenticationDetails(executionContext, providerDefinitionId);
+            var schema = config.GetSchema();
+            var container = new Container(id, StreamMode);
+            var tasks = new List<Task> { emptyTable(container.PrimaryTable, "Data") };
+            foreach (var table in container.Tables)
+                if (await CheckTableExists(executionContext, providerDefinitionId, table.Value.Name.GetValue()))
+                    tasks.Add(emptyTable(table.Value.Name, table.Key));
+
+            await Task.WhenAll(tasks);
 
-            async Task emptyTable(string name, string context)
+            async Task emptyTable(SanitizedSqlName tableName, string context)
```

Comment:
Agree, same as above
"agree about separate internal methods or even just a straightforward method body instead of local functions. Unfortunately haven't had time to properly refactor this.
At least not in scope of this task."